dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -452,6 +477,71 @@ public void downloadZipFileTestAuth() {
         otherUserWorkflowApi.getWorkflowZip(workflowId, versionId);
     }
 
+    /**
+     * Downloads and verifies dockstore-testing/gatk-sv-clinical, a complex WDL workflow with lots
+     * of imports
+     */
+    @Test
+    public void downloadZipComplex() throws IOException {
+        final ApiClient ownerWebClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi ownerWorkflowApi = new WorkflowsApi(ownerWebClient);
+        Workflow refresh = registerGatkSvWorkflow(ownerWorkflowApi);
+
+        Long workflowId = refresh.getId();
+        Long versionId = refresh.getWorkflowVersions().get(0).getId();
+
+        // Try downloading unpublished
+        // Owner: Should pass
+        ownerWorkflowApi.getWorkflowZip(workflowId, versionId);
+
+        // Unfortunately, the generated Swagger client for getWorkflowZip returns void.
+        // Verify the zip contents by making the server side calls
+        final io.dockstore.webservice.core.Workflow dockstoreWorkflow = workflowDAO.findById(workflowId);
+        final Optional<io.dockstore.webservice.core.WorkflowVersion> version = dockstoreWorkflow.getWorkflowVersions().stream()
+                .filter(wv -> wv.getId() == versionId).findFirst();
+        Assert.assertTrue(version.isPresent());
+        final File tempFile = File.createTempFile("dockstore-test", ".zip");
+        try {
+            final FileOutputStream fos = new FileOutputStream(tempFile);
```

Comment:
does fos need to be closed? If so, just move it up to the try

Reply:
It is closed by the method on line 506, but that's a side affect, so moving it up to be explicit